client/src/views/Admin/AdminAuthors.vue

```diff
@@ -28,113 +26,119 @@
-              typeForm="create"
-              v-model:first_name="data.first_name"
-              v-model:last_name="data.last_name"
-              :dataForm="data"
-              path="books/author"
-              :callback="this.getAuthors"
-              :textMessage="`${data.first_name} ${data.last_name} created`"
-              @resetForm="resetForm"
-              @closeModal="closeModal"
-              @showMessage="showMessage"
-              @showErrorMessage="showErrorMessage"
-            />
-          </template>
-        </modal-form>
-
-        <modal-form
-          modal-title="Update record"
-          :displayModal="displayEditModal"
-          @close="closeEditModal"
-        >
-          <template #modal-content>
-            <admin-author-form
-              typeForm="update"
-              v-model:first_name="editForm.first_name"
-              v-model:last_name="editForm.last_name"
-              :dataForm="editForm"
-              :path="`/books/updateauthor/${editForm._id}`"
-              :callback="this.getAuthors"
-              @resetForm="resetEditForm"
-              @closeModal="closeEditModal"
-              @showMessage="showMessage"
-              @showErrorMessage="showErrorMessage"
-            />
-          </template>
-        </modal-form>
+          <admin-author-form
+            @submitForm="onSubmit"
+            :initialForm="initialForm"
+          />
+        </Dialog>
       </template>
     </admin-table>
+    <app-loader v-else />
   </div>
 </template>
 
 <script>
-import AdminTable from "@/components/Admin/AdminTable.vue";
-import ModalForm from "@/components/UI/ModalForm";
+import AdminTable from "@/components/Admin/AdminTable";
+
 import AdminAuthorForm from "@/components/Admin/Forms/AdminAuthorForm";
-import adminFormMixin from "@/mixins/adminFormMixin.js";
-import dataStore from "@/mixins/dataStore.js";
-import toggle from "@/mixins/toggle.js";
+import AppLoader from "@/components/AppLoader";
+
+import { ref, onMounted } from "vue";
+import useSearchedAuthors from "@/hooks/Author/useSearchedAuthors";
+import useForm from "@/hooks/useForm";
+import useData from "@/hooks/useData";
+import useDialog from "@/hooks/useDialog";
+import useMessage from "@/hooks/useMessage";
+
 export default {
   name: "admin-authors",
-  mixins: [toggle, adminFormMixin, dataStore],
+
   components: {
     AdminTable,
-    ModalForm,
     AdminAuthorForm,
+    AppLoader,
   },
-  data() {
+
+  setup() {
+    const {
+      data: authors,
+      loading,
+      error,
+      responseMessage,
+      getData,
+      updateItem,
+      removeItem,
+      createItem,
+    } = useData();
+    const { searchQuery, searchedAuthors } = useSearchedAuthors(authors);
+    const { submitted, displayDialog, hideDialog, showDialog } = useDialog();
+    const { showErrorMessage, showSuccessfulMessage } = useMessage();
+
+    const initialForm = ref({});
+
+    onMounted(() => {
+      getData("/books/allauthors");
```

Comment:
I think better call api functions instead of pass text urls to function parameters.